tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/RetryConditionTest.java

```diff
@@ -215,4 +227,144 @@ public void testClassLevelRetryWithAbortOnTrue() {
+     * ServiceD uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and chains 2 CompletableFutures.
+     */
+    @Test
+    public void testRetryChainSuccess() {
+        final CompletionStage<String> future = asyncRetryClient.serviceD();
+
+        assertCompleteOk(future, "Success");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServD());
+    }
+
+    /**
+     * Persistent Error condition. Will retry 3 times and still throw exception.
+     * ServiceE will always return IOException.
+     */
+    @Test
+    public void testRetryChainExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceE();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(3, asyncRetryClient.getCountInvocationsServE());
+    }
+
+    /**
+     * Persistent Error condition. Will retry 3 times and still throw exception.
+     * ServiceF will always return IOException.
+     */
+    @Test
+    public void testRetryParallelExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceG();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(3, asyncRetryClient.getCountInvocationsServG());
+    }
+
+    /**
+     * Temporary error. Will retry 3 times, the first 2 will fail in a CompletableFuture parallel execution.
+     * ServiceG uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and 2 CompletableFutures.
+     */
+    @Test
+    public void testRetryParallelSuccess() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceG();
+
+        assertCompleteOk(future, "Success then Success");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServG());
+    }
+
+    private void assertCompleteExceptionally(final CompletionStage<String> future,
+                                             final Class<? extends Throwable> exceptionClass,
+                                             final String exceptionMessage) {
+        try {
+            future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS);
+            fail("We were expecting an exception: " + exceptionClass.getName() + " with message: " + exceptionMessage);
+        }
+        catch (InterruptedException | TimeoutException e) {
+            fail("Unexpected exception" + e);
+        }
+        catch (ExecutionException ee) {
+            Assert.assertTrue(exceptionClass.isInstance(ee.getCause()), "Cause of ExecutionException was " + ee.getCause());
+            assertEquals(exceptionMessage, ee.getCause().getMessage());
+        }
+    }
+
+    private void assertCompleteOk(final CompletionStage<String> future, final String expectedMessage) {
+        try {
+            assertEquals(expectedMessage, future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS));
```

Comment:
This wait is too short because `Retry` has a default jitter of 200ms.

Could we increase the timeout to a couple of seconds? (a working implementation should never hit this timeout anyway, so I think a slightly longer wait is justified)